### PR TITLE
[BUGFIX] Remember current template id

### LINKED ISSUE #99: Error "Imported page does not exist"
This error occurs using latest version from TER 2.0.1 when using `sourceFile` and `importPage` and when a PDF is created containing multiple pages. This issue is probably related to #41.

One problem is that there is broken code in
https://github.com/bithost-gmbh/pdfviewhelpers/blob/master/Classes/Model/BasePDF.php#L199

`$this->tpl` is never defined. A strict comparison against `0` is always `true`. This leads to calling `$this->useTemplate(NULL);`, subsequently causing the mentioned error.

### PR DESCRIPTION
**Pull request by liayn**
Caveat: A template with multiple pages will only use the first page of it

Fixes #99

### CODE REVIEW

**Comment by liayn** [comment]
A really dirty workaround

File: Classes/Model/BasePDF.php [hunk 1]
```diff
@@ -196,11 +201,16 @@ public function AddPage($orientation = '', $format = '', $rotationOrKeepmargins
...
         }
 
-        if ($this->importTemplateOnThisPage && $this->tpl !== 0) {
-            $this->useTemplate($this->tpl);
+        if ($this->importTemplateOnThisPage && $this->currentTemplateId) {
+            $this->useTemplate($this->currentTemplateId);
         }
     }
 
+    public function setCurrentTemplateId(int $templateId)
```
Comment (maechler) [inline Classes/Model/BasePDF.php]: Please do not use type hints for primitive data types as we still support PHP 5.6.

File: Classes/Model/BasePDF.php [hunk 2]
```diff
@@ -96,6 +96,11 @@ class BasePDF extends Fpdi
      */
     protected $currentPageFormat = '';
 
+    /**
+     * @var int
+     */
+    protected $currentTemplateId = 0;
```
Comment (maechler) [inline Classes/Model/BasePDF.php]: Isn't this variable already declared in `FpdfTplTrait`?

File: Classes/Model/BasePDF.php (hunk 1 above)
Comment (liayn) [reply Classes/Model/BasePDF.php]: > we still support PHP 5.6

I guess only for December, right? ;-)

**Comment by maechler** [comment]
Thanks for the pull request!

> Caveat: A template with multiple pages will only use the first page of it

If I get this correctly, that means that it works just as before? If you use `importPage` a second time on a different page that should work, or am I wrong? 
I do not think that multiple pages would have been imported automatically anyways, although I have to admit I never tested this use case.

File: Classes/Model/BasePDF.php (hunk 2 above)
Comment (liayn) [reply Classes/Model/BasePDF.php]: No it is not. PHPStorm would have told me.

**Comment by liayn** [comment]
Updated the patch to be really working now and without typehints.

File: Classes/Model/BasePDF.php (hunk 2 above)
Comment (maechler) [reply Classes/Model/BasePDF.php]: Ah, I see..a different trait is included, sry!

File: Classes/Model/BasePDF.php (hunk 1 above)
Comment (maechler) [reply Classes/Model/BasePDF.php]: We are thinking about dropping support for PHP < 7 next year, yes. But if we do so, we would also like to add type hints to all functions and that would be considered a breaking change. Thus I am not yet sure how fast we will make this transition.

**Comment by maechler** [comment]
Thanks, I will try to test and release this as soon as possible!

**Comment by liayn** [comment]
This is a workaround. I'm not sure if this is the right thing to release actually. But I don't know how the inner stuff in fpdf works, so I guess you know better.

**Pull request merged**